src/clic-config-edit.adb

```diff
@@ -0,0 +1,184 @@
+with Ada.Text_IO;
+with Ada.Directories;
+
+with AAA.Strings;
+
+with CLIC.Config.Load;
+
+with Simple_Logging;
+
+with TOML;
+
+package body CLIC.Config.Edit is
+
+   package Trace renames Simple_Logging;
+
+   use TOML;
+
+   procedure Write_Config_File (Table : TOML_Value; Path : String)
+     with Pre => Table.Kind = TOML_Table;
+
+   function Remove_From_Table (Table : TOML_Value;
+                               Key   : Config_Key)
+                               return Boolean
+     with Pre => Table.Kind = TOML_Table;
+
+   function Add_In_Table (Table : TOML_Value;
+                          Key   : Config_Key;
+                          Val   : TOML_Value)
+                          return Boolean
+     with Pre => Table.Kind = TOML_Table;
+
+   -----------------------
+   -- Write_Config_File --
+   -----------------------
+
+   procedure Write_Config_File (Table : TOML_Value; Path : String) is
+      use Ada.Text_IO;
+      use Ada.Directories;
+      File : File_Type;
+   begin
+
+      --  Create the directory for the config file, in case it doesn't exists
+      Create_Path (Containing_Directory (Path));
+
+      Create (File, Out_File, Path);
+      Trace.Debug ("Write config: '" & TOML.Dump_As_String (Table) & "'");
+      Put (File, TOML.Dump_As_String (Table));
+      Close (File);
+   end Write_Config_File;
```

Comment:
I think running this when out of space will result on losing your whole config, so I'd use `AAA.Directories.Replacer` here.